src/seedu/addressbook/data/person/Address.java

```diff
@@ -27,30 +35,87 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+        this.block = assignBlock(trimmedAddress);
+        this.street = assignStreet(trimmedAddress);
+        this.unit = assignUnit(trimmedAddress);
+        this.postalCode = assignPostalCode(trimmedAddress);
+
+    }
+
+    /*
+     * Returns the Person's block of his address.
+     *
+     * @param trimmedAddress the String to be operated on
+     */
+    public Block assignBlock(String trimmedAddress) {
```

Comment:
Consider using a different name for this method, as the block is not in fact assigned, but rather extracted out from the trimmed address.